### PR TITLE
fail in borg package if version metadata is completely broken

### LINKED ISSUE #2538: Very confusing error message when _version.py contains misdetected version
When installing borg from a git repository without tags or when setuptools_scm can not get the correct version for some other reason, borg fails with very confusing error messages.

```
borg list: error: argument REPOSITORY_OR_ARCHIVE: invalid validator value: '/tmp/borg-repo'
```

*How to reproduce:*
easiest way without messing up the repository is to replace _version.py with
```
version = '0.1.dev1+aaaa'
```
But installing from a git repository without tags should reproduce this too.

*Cause:*
While parsing arguments `replace_placeholders` tries to use the version, but throws an exception which is somehow replaced by a confusing message in the argument parser exception handling.
```
>>> borg.helpers.Location('/tmp/borg-repo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/martin/sw-dev/borg/src/borg/helpers.py", line 979, in __init__
    if not self.parse(self.orig):
  File "/home/martin/sw-dev/borg/src/borg/helpers.py", line 983, in parse
    text = replace_placeholders(text)
  File "/home/martin/sw-dev/borg/src/borg/helpers.py", line 696, in replace_placeholders
    'borgpatch': '%d.%d.%d' % borg_version_tuple[:3],
TypeError: %d format: a number is required, not str
```

*Possible solution:*
Do some sanity checking in `borg/__init__.py` and abort early with a clear error message?

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this helps to fail early when people do badly done scm based repackaging

this should resolve  #2538

### CODE REVIEW

File: src/borg/__init__.py [hunk 1]
```diff
@@ -4,3 +4,13 @@
...
 __version_tuple__ = tuple(LooseVersion(__version__).version[:3])
+
+# assert that all semver components are integers
+# this is mainly to show errors when people repackage poorly
+# and setuptools_scm determines a 0.1.dev... version
+assert all(isinstance(v, int) for v in __version_tuple__), \
+    """\
+broken borgbackup version metadata: %r
+
+please ensure the real version metadata is availiable
```
Comment (textshell) [inline src/borg/__init__.py]: maybe?
please ensure the real version metadata is availiable in _version.py

Comment (ThomasWaldmann) [inline src/borg/__init__.py]: typo "available"

also: that "please ensure ..." is not very helpful if the reader does not know how to ensure that.

Comment (ThomasWaldmann) [inline src/borg/__init__.py]: @RonnyPfannschmidt can you update/fix it, so we can merge it?

File: src/borg/__init__.py [hunk 2]
```diff
@@ -4,3 +4,15 @@
...
+# assert that all semver components are integers
+# this is mainly to show errors when people repackage poorly
+# and setuptools_scm determines a 0.1.dev... version
+assert all(isinstance(v, int) for v in __version_tuple__), \
+    """\
+broken borgbackup version metadata: %r
+
+version metadata is obtained dynamically on installation via setuptools_scm,
+please ensure your git repo has the correct tags or you provide the version
+using SETUPTOOLS_SCM_PRETEND_VERSION in your build script
```
Comment (ThomasWaldmann) [inline src/borg/__init__.py]: add a dot at the end of the sentence, please.

Comment (RonnyPfannschmidt) [inline src/borg/__init__.py]: done

**Comment by ThomasWaldmann** [comment]
thanks!

**Pull request merged**